dotnet/src/Connectors/Connectors.Ollama/Extensions/OllamaServiceCollectionExtensions.cs

```diff
@@ -110,10 +111,20 @@ public static IServiceCollection AddOllamaChatCompletion(
         Verify.NotNull(services);
 
         services.AddKeyedSingleton<IChatCompletionService>(serviceId, (serviceProvider, _) =>
-            new OllamaChatCompletionService(
-                modelId: modelId,
-                endpoint: endpoint,
-                loggerFactory: serviceProvider.GetService<ILoggerFactory>()));
+        {
+            var ollamaClient = new OllamaApiClient(endpoint, modelId);
+            var logger = serviceProvider.GetService<ILoggerFactory>()?.CreateLogger(ollamaClient.GetType());
+
+            var chatClientBuilder = new ChatClientBuilder()
+                .UseFunctionInvocation();
+
+            if (logger is not null)
+            {
+                chatClientBuilder.UseLogging(logger);
+            }
```

Comment:
Might want to use UseOpenTelemetry instead (or possibly in addition, though it'll be somewhat duplicative).

Reply:
Will keep logging and if there's asks, may consider adding a flag to use telemetry in the extension.

Reply:
Doesn't that differ from the existing connectors?